cpp/src/arrow/flight/sql/odbc/odbc_impl/system_dsn.h

```diff
@@ -19,6 +19,7 @@
 #include "arrow/flight/sql/odbc/odbc_impl/platform.h"
 
 #include "arrow/flight/sql/odbc/odbc_impl/config/configuration.h"
+#include "arrow/result.h"
```

Comment:
nit: if you wanted arrow::Status, just include status.h

Reply:
Done.